sqlglot/dialects/tsql.py

```diff
@@ -37,11 +106,44 @@ class Parser(Parser):
             "CHARINDEX": exp.StrPosition.from_arg_list,
         }
 
-        def _parse_convert(self):
+        var_length_datatype = [
+            DataType.Type.NVARCHAR,
+            DataType.Type.VARCHAR,
+            DataType.Type.CHAR,
+            DataType.Type.NCHAR,
+        ]
+
+        def _parse_convert(self, strict):
             to = self._parse_types()
             self._match(TokenType.COMMA)
             this = self._parse_field()
-            return self.expression(exp.Cast, this=this, to=to)
+
+            # Retrieve length of datatype and override to default if not specified
+            if list_get(to.expressions, 0) is None and to.this in self.var_length_datatype:
+                to = exp.DataType.build(to.this, expressions=[exp.Literal.number(30)], nested=False)
+
+            # Check whether a conversion with format is applicable
+            if self._match(TokenType.COMMA):
+                format_norm = exp.Literal.string(TSQL.convert_format_mapping[int(self._parse_number().this)])
```

Comment:
int(self._parse_number().name)

Reply:
why convert to int and just make the format_mapping as tring?

Reply:
True, did just that now. Also built in a check on the validity of the users entry.